pkg/stub/handler.go

```diff
@@ -331,23 +421,43 @@ func (h *Handler) IsRetryableAPIError(err error) bool {
 	return false
 }
 
-func (h *Handler) CreateDefaultResourceIfNeeded() error {
+func (h *Handler) CreateDefaultResourceIfNeeded() (*v1alpha1.SamplesResource, error) {
 	// coordinate with event handler processing
 	// where it will set h.sampleResource
 	// when it completely updates all imagestreams/templates/statuses
 	h.mutex.Lock()
 	defer h.mutex.Unlock()
-	if h.waitingForCredential != nil {
-		return nil
-	}
 
 	deleteInProgress := h.deleteInProgress
 
 	var err error
-	var srcfg *v1alpha1.SamplesResource
+	// "4a" in the "startup" workflow, just create default
+	// resource and set up that way
+	srcfg := &v1alpha1.SamplesResource{}
+	srcfg.Name = v1alpha1.SamplesResourceName
+	srcfg.Kind = "SamplesResource"
+	srcfg.APIVersion = "samplesoperator.config.openshift.io/v1alpha1"
```

Comment:
get this out of pkg/apis/samplesoperator/v1alpha1/register.go

Reply:
OK ... I will change the private constants the sdk setup command auto generated to public and do this